### DISCUSSION TITLE
Exporting (pdf, ePub)

### DISCUSSION THREAD

**Discussion by reneaudet** [knowledgefutures/pubpub#1962] [Support]
A few questions about exporting.

- Is there a way that the exported files (pdf or ePub) take into account custom CSS?
- I have lot of problems exporting in ePub (I received the message "There was a problem generating the file" or the ePub file is unreadable, even less qualificating in ePub validator).
- I have inconsistencies in the PDF rendering (paragraphs missing, unequal spacing between footnotes, etc.)

Is it "normal"? Are there types of content or styling better to avoid?

Thanks a lot!

**Comment by gabestein** [comment]
Hello! I'm sorry you've been experiencing these issues. At the moment, we do not offer customization for autogenerated exports. It's something we would consider in the future, but at the moment the CSS is very complicated, difficult to debug and prone to error, and as such not something we have the bandwidth to support.

Additionally, PubPub is meant to be a web-native platform, and though we know offering PDF support is important, our offering is meant to be "good enough." Autogeneration is simply never going to yield perfect results due to the inherent differences between browser and PDF rendering. For folks who need customization of these assets, we recommend downloading versions of the final pubs in an intermediary format and working with a compositor to create custom versions.

We just pushed some updates that should fix the validation issues with ePubs and docx, but let us know if you still run into issues (to produce new versions, you will need to regenerate the ePubs by either making changes to the draft or re-releasing the pubs).

All that said, we would of course like to make the PDFs we autogenerate as good as possible, and I'm sorry you're having issues with them. If you can send us examples of Pubs, I'd like to take a look at those bugs like missing paragraphs and footnote spacing issues.
